### ISSUE TITLE
Publish dataset after editing a file 

### ISSUE THREAD

**Issue by sbarbosadataverse** [IQSS/dataverse#4153]
When on a file landing page and you create a new version aka "draft," there is no way to publish. 

see RT ticket for example: https://help.hmdc.harvard.edu/Ticket/Display.html?id=253546 

Note that there is a work-around: user can go to the dataset page, select the file, and select the new version from the version tab on the file page. Then publish appears on the draft file page.

**Comment by TaniaSchlatter** [comment]
Steps to recreate:

On the page of a published dataset:
1. select a file -> go to file page
2. Edit the file – restrict/unrestrict/uploading new – any edit option except delete
3. A draft is created, user is on the new draft page
4. There is not a button to publish the new draft.

**Comment by TaniaSchlatter** [comment]
Note per discussion with @scolapasta: user who wants to publish a file after making an edit needs to know that publishing the change to the file re-publishes the dataset, and creates a new version of the dataset. On publish, user needs to select if it is a major or minor version change.

**Comment by jggautier** [comment]
> user who wants to publish a file after making an edit needs to know that publishing the change to the file re-publishes the dataset, and creates a new version of the dataset.

This feels like a bandaid, but maybe the success message can include something like: "When you're ready, `<hyperlink>`visit the dataset page`</hyperlink>` to publish your changes." ("the dataset page" would point to the draft version of the dataset page)

The big problem is that this might be contributing to more draft datasets that are never published, or published later than necessary, because depositors don't know how to publish their changes. I'm wondering if measuring the effect of this problem could include seeing how long draft datasets remain drafts.

**Comment by TaniaSchlatter** [comment]
Based on results of the small Harvard Dataverse repository Publish Workflow study, #109, it would be somewhat effective to add a message at the top of the page to publish, especially if the format was different from the standard green banner, which participants tended not to see. It is likely to be very effective to add the "publish" button to the file page.

**Comment by mreekie** [comment]
This has been a UX issue that has been a blocker for publishing.  This is the situation where if you change metadata on the file page. There is no publish button and no indication that you need to publish. 

User testing has shown that having the publish button is more effective than having the message at the Top.

**Comment by mreekie** [comment]
sprint:
Some discussion. 
- Would button only be present if a file change triggers a need to publish the dataset?
- What if the action of publishing went in the message itself?
- What about not everyone who can edit can publish?  Needs to be dynamic

- medium.

**Comment by qqmyers** [comment]
Another thought - we also have a problem with people making a dataset level change and not realizing they have to publish - would it work to just make the 'draft' tag more clear on both dataset and file pages i.e. if/when you're looking at draft info there's a bigger banner/message that says you have to publish to make the change visible?

**Comment by qqmyers** [comment]
~partly. My main thought was to have the dataset and file page work the same way, so a change similar to #7675 to update the success message when you edit a file would do that - so perhaps adding a similar change to the file page success message is an alternative to putting a publish button on the file page. 

That said, the success messages are shown only once, so if part of the goal with a publish button is to be a persistent reminder, perhaps a tag that says 'Draft - latest updates are not yet public' or similar would keep the reminder as people move between dataset and file pages.

(FWIW: I'm concerned that publishing the dataset from the file page, when the dataset and other files may have changes that you can't see from that page is going to be confusing, as would deciding whether there should be a publish button one one file page if there are dataset metadata or changes to other files but not the one you're looking at. Doing something with the messaging about the draft status seems clearer to me, hence the suggestions).

**Comment by pdurbin** [comment]
I just went through the "steps to recreate" at https://github.com/IQSS/dataverse/issues/4153#issuecomment-332268214 (thanks @TaniaSchlatter) and made a few screenshots as a visual reference. I chose "Restrict" as the way to create a new draft of the dataset from the file page.

## Screenshot: About to click "Restrict"

Note that "Version 1.0" or "V1" appears in various places. Users probably have no idea that this will soon undergo a subtle change.

![Screen Shot 2022-04-28 at 4 40 56 PM](https://user-images.githubusercontent.com/21006/165846234-83ad5d89-e9fe-4f7d-a181-2cf91e8aa629.png)

## Screenshot: Restrict popup

I'm just including this for completeness. The rest of the page looks the same as above

![Screen Shot 2022-04-28 at 4 41 06 PM](https://user-images.githubusercontent.com/21006/165846258-f5b18856-c8a1-4e9e-bc89-2d61c8f4b4c3.png)

## Screenshot: Success

Here we are probably hoping that the user notices that "Version 1.0" has changed to "Draft" and "V1" has changed to "DRAFT VERSION". (SHOUTING HELPS, right?) 

The success message is not particularly informative (more on this later). And as @TaniaSchlatter points out in the research above, people are somewhat blind to these green success messages.

What is the user supposed to do next? They have to click the title of their dataset ("dogs" in the screenshot) and find the publish button there. This seems like a lot to ask, especially for anyone new to Dataverse.

![Screen Shot 2022-04-28 at 4 41 12 PM](https://user-images.githubusercontent.com/21006/165846307-81865070-84fb-40a9-82a0-cba1fa97bdcf.png)

## Thoughts

To most obvious quick fix (already suggested above) is to modify the message. Perhaps it could say:

"Success! - The file has been updated. Don't forget to [publish]() your dataset." where "publish" could simply be a link to the dataset landing page where there is a prominent publish button. 

While in draft, I do think adding some sort of permanent message or button would make sense. Something to help the user navigate to the dataset page. Something less subtle to indicate that the dataset is still unpublished. And if it's a message, maybe it should also apply to the dataset page, as @qqmyers suggests. Needs more thought.

**Comment by TaniaSchlatter** [comment]
I've mocked up a few options for the case of not adding a "Publish/submit for Review" button to the file page and taking @qqmyers's comment in to account [on the Miro board.](https://miro.com/app/board/o9J_leCZVUU=/?moveToWidget=3458764524329903869&cot=14)

![Screen Shot 2022-05-02 at 1 36 46 PM](https://user-images.githubusercontent.com/15386424/166296694-8c64783e-cd8b-4c91-8176-73e1c1231737.png)

**Comment by mreekie** [comment]
Prioritization:
(Tania)
 - Publish button on the file page
   - Tania fleshed out editing a message/provide a link to dataset apge
   - Jim is suggesting make the "Draft" tag clear, but this may not be enought to get users to make the connection.
   - Sonia needs input from Sonia.
   - Sounds like the meeting will be tomorrow: Sonia, Gustavo, Tania at 4PM. Tania will call the meeting. They will decide on the next steps are in the design.

**Comment by qqmyers** [comment]
Fwiw: option 2 could be broader, e.g. be a general warning on a draft dataset that it isn't live (versus being specific to file changes).

**Comment by TaniaSchlatter** [comment]
Outcomes from discussing with @sbarbosadataverse and @scolapasta:

On the **dataverse** page:
- Return to the basic "Success" message
- Add a "warning" message on the dataset page when one or more files in a dataset have been edited.
- Add a variation on the "warning" message to prompt for submitting for review.
- Keep the warning message until the dataset is published or submitted for review.
![Screen Shot 2022-05-06 at 1 27 14 PM](https://user-images.githubusercontent.com/15386424/167182536-caa46a98-a3f7-4720-85f0-508d0140f712.png)

On the **file** page:
![Screen Shot 2022-05-06 at 1 27 07 PM](https://user-images.githubusercontent.com/15386424/167182537-65887afd-a9fd-4ba6-96ed-c5d5a155871c.png)
- Keep the current basic Success! message
- Add a "warning" message to the file page with link to the dataset when the file has been edited. 
- Add a variation on the "warning" message when submitting for review.
- Keep the warning message until the dataset is published or submitted for review.

**Comment by sekmiller** [comment]
If there are dataset changes to the metadata or terms, etc. (not files), the do we want these warning messages?

Also, is there a difference in the messages if the dataset is unpublished?

**Comment by sekmiller** [comment]
<img width="1164" alt="Screen Shot 2022-05-10 at 3 01 51 PM" src="https://user-images.githubusercontent.com/2364928/167702950-a5fea008-c4d6-49ec-8957-4ead8b4dcafe.png">

Sonia suggested this wording for the Submit for Review version of the message.

**Comment by jggautier** [comment]
Or "please submit it for review to notify the Dataverse collection admin"? I don't think the collection will always be a "journal" Dataverse collection, right?

**Comment by sekmiller** [comment]
Another thing to weigh in on....
<img width="844" alt="Screen Shot 2022-05-10 at 3 11 21 PM" src="https://user-images.githubusercontent.com/2364928/167704522-57a48b88-5c7d-43aa-8c34-03db0fb0240e.png">

**Comment by sekmiller** [comment]
@sbarbosadataverse, @jggautier - do we like/hate/have suggested changes for the enhanced reminder tags on the dataset cards?

**Comment by jggautier** [comment]
I think the text in the light blue reminder tag will be helpful.

The dark and light blue tags seem repetitive to me. They would always always appear together, right? I don't have all of the context so I'm worried I'm not considering everything here.

**Comment by scolapasta** [comment]
1. Agree with Julian, it's repetitive. Personally, I don't like it - these are supposed to be short labels. If anything, maybe have the text on it, if you hover over it? I don't know, feels out of scope for what we already have here and better to do what we already have / agreed upon and see if behavior improves.

**Comment by sekmiller** [comment]
I agree that it is beyond the original scope. It was a suggestion from Sonia during my demo. She says that people miss the "Draft" tag or don't know what it means.

**Comment by pdurbin** [comment]
+1 for hover text for the "Draft" label.

Here is the terminology we agreed on for 4.0 from the "Dataverse & Dataset Terminology" doc at https://docs.google.com/document/d/1TEVgdxwJE4KWZ9Zaus1qGmD0yOiMsW9XKZL0s0NsTl4/edit?usp=sharing

- unpublished dataset: a dataset that  contains only one version in draft state, i.e. has never been published
- published dataset: a dataset that contains at least one version that has been published
- unpushed dataset version: a published dataset that has a new version in draft state
- draft dataset version: a dataset that has a version in draft state, i.e.. an unpublished dataset or a unpushed dataset version

(You'll see "pushed" throughout that doc but obviously, this term never took off!)

I'll also plug the glossary issue:

- #4274

**Comment by jggautier** [comment]
+1 to "and see if behavior improves"! Will this be tested with users after it makes it's way to repositories? Maybe we could see how this change affects the number of unpublished versions or how long they stay unpublished?
